api/lib/utils.js

```diff
@@ -115,6 +117,34 @@ function isValidLedgerWord(ledger) {
   return (/^current$|^closed$|^validated$/.test(ledger));
 }
 
+function attachDate(api, baseTransactions, callback) {
+  var groupedTx = _.groupBy(baseTransactions, function(tx) {
+    return tx.ledger_index;
+  });
+
+  async.each(_.keys(groupedTx), function(ledger, next) {
```

Comment:
Also, I can read this fine. If this is necessary for this PR I can do it, otherwise it's blocking me